src/rdb.c

```diff
@@ -1214,6 +1214,27 @@ ssize_t rdbSaveSingleModuleAux(rio *rdb, int when, moduleType *mt) {
     return io.bytes;
 }
 
+int functionsSaveRio(rio *rdb) {
+    dict *functions = functionsGet();
+    dictIterator *iter = dictGetIterator(functions);
+    dictEntry *entry = NULL;
+    while ((entry = dictNext(iter))) {
+        rdbSaveType(rdb, RDB_OPCODE_FUNCTION);
+        functionInfo* fi = dictGetVal(entry);
+        if (rdbSaveRawString(rdb, (unsigned char *) fi->name, sdslen(fi->name)) == -1) return C_ERR;
+        if (rdbSaveRawString(rdb, (unsigned char *) fi->ei->name, sdslen(fi->ei->name)) == -1) return C_ERR;
+        if (fi->desc) {
+            if (rdbSaveLen(rdb, 1) == -1) return C_ERR; /* desc exists */
+            if (rdbSaveRawString(rdb, (unsigned char *) fi->desc, sdslen(fi->desc)) == -1) return C_ERR;
+        } else {
+            if (rdbSaveLen(rdb, 0) == -1) return C_ERR; /* desc not exists */
+        }
+        if (rdbSaveRawString(rdb, (unsigned char *) fi->code, sdslen(fi->code)) == -1) return C_ERR;
+    }
+    dictReleaseIterator(iter);
```

Comment:
If returns `C_ERR`, `dictReleaseIterator` will not be called.

Reply:
Fixed, not resolving so you can review the changes.